### PR TITLE
GA for Invoicing APIs

### PR DESCRIPTION
**Pull request by bradenwatkins**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

#### Invoice
- Return void invoices by default

#### Transaction
- Modified non-documented flags from query parameters to headers
    - GetChargesFromLegacy -> x-ms-get-charges-from-legacy
    - FilterRefundEligible -> x-ms-filter-refund-eligible
- Modified transactions by customer scope to include billingProfile to match overarching customer scope for this version
- Remove /transactionsDownload endpoint in favor of existing /transactions/download endpoint.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json [hunk 1]
```diff
@@ -80,860 +93,1224 @@
     },
     "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/downloadDocuments": {
```
Comment (ramkd12) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json:94]: Downloading Invoice api, what do they take?

I believe in past we used to take url etc.

other than InvoiceId, nothing else should be present in the request body.

Also, do we have a singular invoice download link?

**Comment by wilcobmsft** [comment]
```
Modified non-documented flags from query parameters to headers
GetChargesFromLegacy -> x-ms-get-charges-from-legacy
FilterRefundEligible -> x-ms-filter-refund-eligible
```

Any reason this isn't reflected in the swagger?

**Comment by wilcobmsft** [comment]
Did we get approval from Breaking Change Review Board for removing /transactionsDownload?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/types.json [hunk 2]
```diff
@@ -84,6 +84,11 @@
       "description": "Common fields that are returned in the response for all Azure Resource Manager resources.",
       "type": "object",
       "properties": {
+        "type": {
+          "description": "The type of the resource. E.g. \"Microsoft.Compute/virtualMachines\" or \"Microsoft.Storage/storageAccounts\".",
+          "type": "string",
+          "readOnly": true
+        },
         "id": {
           "description": "Fully qualified resource ID for the resource. Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}.",
```
Comment (mumami) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/types.json:93]: Should we call this as ResourceId or ResourceUri instead? If we have been using Resource ID then let's continue with the same

Comment (bradenwatkins) [reply specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/types.json:93]: This should stay as Resource ID (and this actually comes from ARM's definition of a resource).

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json [hunk 3]
```diff
@@ -32,28 +376,183 @@
...
+          "TransactionsListByInvoice": {
+            "$ref": "./examples/transactionsListByInvoice.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/invoices/{invoiceName}/transactionSummary": {
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json:423]: Should this be `/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/invoices/{invoiceName}/transactionSummary/default` as it is not a "List"?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json [hunk 4]
```diff
@@ -80,860 +93,1247 @@
...
+          "InvoicesDownloadDocumentsByBillingAccount": {
+            "$ref": "./examples/invoicesDownloadDocumentsByBillingAccount.json"
+          }
         }
       }
     },
-    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/invoices": {
-      "get": {
-        "description": "Lists the invoices for a billing account for a given start date and end date. The operation is supported for billing accounts with agreement type Microsoft Partner Agreement or Microsoft Customer Agreement.",
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/invoices/{invoiceName}/amend": {
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json:167]: Understand why this was named "amend"

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json [hunk 5]
```diff
@@ -80,860 +93,1247 @@
...
       }
     },
-    "/providers/Microsoft.Billing/billingAccounts/default/billingSubscriptions/{subscriptionId}/invoices": {
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/invoices": {
       "get": {
-        "description": "Lists the invoices for a subscription.",
+        "tags": [
+          "Invoice"
+        ],
+        "description": "Lists the invoices for a billing account for a given start date and end date. The operation is supported for billing accounts with agreement type Microsoft Partner Agreement or Microsoft Customer Agreement.",
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json:403]: MOSP if you are an account owner

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json [hunk 6]
```diff
@@ -80,860 +93,1247 @@
...
         }
       }
     },
     "/providers/Microsoft.Billing/billingAccounts/default/billingSubscriptions/{subscriptionId}/invoices/{invoiceName}/download": {
       "post": {
-        "description": "Gets a URL to download an invoice.",
+        "tags": [
+          "Invoice"
+        ],
+        "description": "ets a URL to download an invoice by billing subscription.",
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json:553]: Fix description here

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json [hunk 7]
```diff
@@ -80,860 +93,1247 @@
...
         "expiryTime": {
           "description": "The time in UTC when the download URL will expire.",
-          "format": "date-time",
-          "readOnly": true,
-          "type": "string"
+          "type": "string",
+          "readOnly": true
         },
         "url": {
           "description": "The URL to the PDF file.",
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json:805]: PDF or Zip file

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json [hunk 8]
```diff
@@ -80,860 +93,1247 @@
...
+          "readOnly": true,
           "x-ms-enum": {
-            "modelAsString": true,
-            "name": "InvoiceDocumentType"
+            "name": "InvoiceDocumentType",
+            "modelAsString": true
           }
         },
+        "chargeDate": {
+          "description": "Upcoming feature.",
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json:1012]: What is this used for? Daily invoicing or something else?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json [hunk 9]
```diff
@@ -80,860 +93,1247 @@
...
+        "amountRequested": {
+          "description": "The amount refunded.",
+          "allOf": [
+            {
+              "$ref": "./types.json#/definitions/Amount"
+            }
+          ]
+        },
+        "amountRefunded": {
+          "description": "The amount of refund requested.",
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/invoice.json:1257]: Fix description here

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json [hunk 10]
```diff
@@ -63,259 +562,377 @@
...
+            }
+          ]
         },
         "exchangeRate": {
+          "format": "double",
           "description": "The exchange rate used to convert charged amount to billing currency, if applicable.",
-          "readOnly": true,
           "type": "number"
         },
         "invoice": {
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json:730]: Investigate behavior of this field for billed and unbilled. Can we improve the description?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json [hunk 11]
```diff
@@ -63,259 +562,377 @@
...
-          "description": "The ID of the invoice section which will be billed for the transaction.",
-          "readOnly": true,
+          "description": "The fully qualified ID that uniquely identifies an invoice section.",
           "type": "string"
         },
+        "isThirdParty": {
+          "description": "Whether or not the transaction is third party.",
+          "type": "boolean"
+        },
         "kind": {
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json:750]: Understand what this field comes from and why the description is different from the values

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json [hunk 12]
```diff
@@ -63,259 +562,377 @@
...
+          "description": "The unit of measure used to bill for the product. For example, compute services are billed per hour.",
+          "type": "string"
+        },
         "unitType": {
           "description": "The description for the unit of measure for a given product.",
-          "readOnly": true,
           "type": "string"
+        },
+        "specialTaxationType": {
+          "description": "Identifies the type of tax calculation used for the invoice. The field is applicable only to invoices with special tax calculation logic.",
```
Comment (bradenwatkins) [inline specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json:854]: Is this country or locale specific?

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json (hunk 10 above)
Comment (bradenwatkins) [reply specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json:730]: This is the "InvoiceName" whereas "InvoiceId" is the fully qualified Arm ID. But we can't change it here.

File: specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json (hunk 11 above)
Comment (bradenwatkins) [reply specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/transaction.json:750]: It's a passthrough from MCAPI line item. We do no work on this property.

**Pull request merged**